test/unit/mem/copy/src/BufSlicing.cpp

```diff
@@ -0,0 +1,308 @@
+/* Copyright 2019 Axel Huebl, Benjamin Worpitz, Jakob Krude
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <alpaka/acc/AccCpuSerial.hpp>
+#include <alpaka/test/acc/TestAccs.hpp>
+#include <alpaka/test/mem/view/Iterator.hpp>
+#include <alpaka/test/Extent.hpp>
+#include <catch2/catch.hpp>
+
+
+template<
+    typename TDim,
+    typename TIdx,
+    typename TAcc,
+    typename TData,
+    typename Vec = alpaka::vec::Vec<
+        TDim,
+        TIdx>>
+struct TestContainer
+{
+
+    using AccQueueProperty = alpaka::queue::Blocking;
+    using DevQueue = alpaka::queue::Queue<
+        TAcc,
+        AccQueueProperty
+    >;
+    using DevAcc = alpaka::dev::Dev<TAcc>;
+    using PltfAcc = alpaka::pltf::Pltf<DevAcc>;
+
+    using DevHost = alpaka::dev::DevCpu;
+    using PltfHost = alpaka::pltf::Pltf<DevHost>;
+
+    using BufHost = alpaka::mem::buf::Buf<
+        DevHost,
+        TData,
+        TDim,
+        TIdx
+    >;
+    using BufDevice = alpaka::mem::buf::Buf<
+        DevAcc,
+        TData,
+        TDim,
+        TIdx
+    >;
+
+    using SubView = alpaka::mem::view::ViewSubView<
+        DevAcc,
+        TData,
+        TDim,
+        TIdx
+    >;
+
+    DevAcc const devAcc;
+    DevHost const devHost;
+    DevQueue devQueue;
+
+
+    // Constructor
+    TestContainer():
+        devAcc(alpaka::pltf::getDevByIdx<PltfAcc>(0u)),
+        devHost(alpaka::pltf::getDevByIdx<PltfHost>(0u)),
+        devQueue(devAcc)
+    {}
+
+
+    auto createHostBuffer(
+        Vec extents,
+        bool indexed
+    ) -> BufHost
+    {
+        BufHost bufHost(
+            alpaka::mem::buf::alloc<
+                TData,
+                TIdx
+            >(
+                devHost,
+                extents
+            ));
+        if(indexed)
+        {
+            TData *const ptr = alpaka::mem::view::getPtrNative(bufHost);
+            for(TIdx i(0);i < extents.prod();++i)
+            {
+                ptr[i] = static_cast<TData>(i);
+            }
+        }
+        return bufHost;
+    }
+
+
+    auto createDeviceBuffer(Vec extents) -> BufDevice
+    {
+        BufDevice bufDevice(
+            alpaka::mem::buf::alloc<
+                TData,
+                TIdx
+            >(
+                devHost,
```

Comment:
BUG: this must be `devHost`

Reply:
Do you mean `devAcc`?

Reply:
ohh yes I repeated the error ^^, it must be `devAcc`